Framework/CurveFitting/src/Algorithms/PlotPeakByLogValue.cpp

```diff
@@ -268,10 +276,16 @@ void PlotPeakByLogValue::exec() {
         if (createFitOutput)
           wsBaseName = wsNames[i].name + "_" + spectrum_index;
 
+        bool histogramFit = getPropertyValue("EvaluationType") == "Histogram";
+
         // Fit the function
         API::IAlgorithm_sptr fit =
             AlgorithmManager::Instance().createUnmanaged("Fit");
         fit->initialize();
+        if (histogramFit) {
+          fit->setPropertyValue("EvaluationType",
+                                getPropertyValue("EvaluationType"));
+        }
```

Comment:
Why not always forward that property value? Would create fewer problems in case the default is changed?